core/src/main/java/cucumber/runtime/RuntimeOptionsFactory.java

```diff
@@ -1,170 +1,244 @@
 package cucumber.runtime;
 
+import static java.util.Arrays.asList;
+import cucumber.api.CucumberConfiguration;
+import cucumber.api.CucumberConfiguration.Type;
 import cucumber.api.CucumberOptions;
+import cucumber.runtime.configuration.YamlConfigurationParser;
 import cucumber.runtime.io.MultiLoader;
-
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
-import static java.util.Arrays.asList;
-
 public class RuntimeOptionsFactory {
-    private final Class clazz;
-    private boolean featuresSpecified = false;
-    private boolean overridingGlueSpecified = false;
-
-    public RuntimeOptionsFactory(Class clazz) {
-        this.clazz = clazz;
-    }
-
-    public RuntimeOptions create() {
-        List<String> args = buildArgsFromOptions();
-        return new RuntimeOptions(args);
-    }
-
-    private List<String> buildArgsFromOptions() {
-        List<String> args = new ArrayList<String>();
-
-        for (Class classWithOptions = clazz; hasSuperClass(classWithOptions); classWithOptions = classWithOptions.getSuperclass()) {
-            CucumberOptions options = getOptions(classWithOptions);
-            if (options != null) {
-                addDryRun(options, args);
-                addMonochrome(options, args);
-                addTags(options, args);
-                addPlugins(options, args);
-                addStrict(options, args);
-                addName(options, args);
-                addSnippets(options, args);
-                addGlue(options, args);
-                addFeatures(options, args);
-                addJunitOptions(options, args);
-            }
+  private static final String FILE_PREFIX = "file:";
+  private static final String CLASSPATH_PREFIX = "classpath:";
+  private final Class<?> clazz;
+  private boolean featuresSpecified = false;
+  private boolean overridingGlueSpecified = false;
+
+  public RuntimeOptionsFactory(Class<?> clazz) {
+    this.clazz = clazz;
+  }
+
+  public RuntimeOptions create() {
+    CucumberConfiguration configuration = clazz.getAnnotation(CucumberConfiguration.class);
+
+    // TODO: evaluate annotations on super classes
+    if (configuration == null || configuration.type().equals(Type.ANNOTATION)) {
+      return new RuntimeOptions(buildArgsFromOptions());
```

Comment:
I am not sure if it is wise to let the `RuntimeOptionsFactory` handle both the `@CucumberConfiguration` and `@CucumberOptions` annotations. They are different sources of configuration that can exist together and should be applied in some order.